_deploy/r/gnoswap/gnft/gnft.gno

```diff
@@ -84,61 +135,59 @@ func Approve(user pusers.AddressOrName, tid grc721.TokenID) {
 	}
 }
 
+// SetApprovalForAll enables or disables approval for a third party to manage all tokens
+// Parameters:
+//   - user: The address or username to set approval for
+//   - approved: true to approve, false to revoke approval
 func SetApprovalForAll(user pusers.AddressOrName, approved bool) {
 	common.IsHalted()
-
-	err := gnft.SetApprovalForAll(users.Resolve(user), approved)
-	if err != nil {
-		panic(err.Error())
-	}
+	checkErr(gnft.SetApprovalForAll(users.Resolve(user), approved))
 }
 
+// TransferFrom transfers ownership of a token from one address to another
+// Parameters:
+//   - from: The current owner's address or username
+//   - to: The recipient's address or username
+//   - tid: The token ID to transfer
 func TransferFrom(from, to pusers.AddressOrName, tid grc721.TokenID) {
 	common.IsHalted()
-
-	err := gnft.TransferFrom(users.Resolve(from), users.Resolve(to), tid)
-	if err != nil {
-		panic(err.Error())
-	}
+	checkErr(gnft.TransferFrom(users.Resolve(from), users.Resolve(to), tid))
 }
 
-// Admin
+// Mint creates a new NFT and assigns it to the specified address (only callable by owner)
+// Parameters:
+//   - to: The address or username to mint the token to
+//   - tid: The token ID to assign to the new NFT
+//
+// Returns:
+//   - grc721.TokenID: The ID of the newly minted token
 func Mint(to pusers.AddressOrName, tid grc721.TokenID) grc721.TokenID {
+	owner.AssertCallerIsOwner()
 	common.IsHalted()
 
-	caller := std.PrevRealm().Addr()
-	assertIsAdmin(caller)
-
-	err := gnft.Mint(users.Resolve(to), tid)
-	if err != nil {
-		panic(err.Error())
-	}
+	checkErr(gnft.Mint(users.Resolve(to), tid))
 
-	userMinted[users.Resolve(to)] = append(userMinted[users.Resolve(to)], tid)
+	toAddr := users.Resolve(to)
+	appendTokenList(toAddr, tid)
 	return tid
 }
 
+// Burn removes a specific token ID (only callable by owner)
+// Parameters:
+//   - tid: The token ID to burn
 func Burn(tid grc721.TokenID) {
+	owner.AssertCallerIsOwner()
 	common.IsHalted()
 
-	caller := std.PrevRealm().Addr()
-	assertIsAdmin(caller)
-	err := gnft.Burn(tid)
-	if err != nil {
-		panic(err.Error())
-	}
-}
-
-func SetAdmin(newAdmin pusers.AddressOrName) {
-	common.IsHalted()
-
-	caller := std.PrevRealm().Addr()
-	assertIsAdmin(caller)
-	admin = users.Resolve(newAdmin)
+	checkErr(gnft.Burn(tid))
```

Comment:
For Burn, the corresponding tokenId should be deleted from the tokenIdList.

Reply:
53464b183d9fcfbf86fc280162eb5b8f86b66da6